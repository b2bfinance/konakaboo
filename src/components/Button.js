import styled, { css } from 'styled-components';

const Button = styled.a`
  align-items: center;
  border-radius: 0.15rem;
  box-sizing: border-box;
  cursor: pointer;
  display: flex;
  font-weight: ${props => props.theme.mainBoldFontWeight};
  justify-content: center;
  padding: 0.75rem 1.5rem;
  text-decoration: none;
  width: 100%;
  white-space: nowrap;
  user-select: none;

  ${props =>
    props.primary &&
    css`
      background-color: ${props.theme.applyButtonBackground};
      color: ${props.theme.applyButtonColor};
    `};

  ${props =>
    props.secondary &&
    css`
      background-color: ${props.theme.infoButtonBackground};
      color: ${props.theme.infoButtonColor};
    `};

  ${props =>
    props.slim &&
    css`
      padding: 0.25rem 0.5rem;
      font-size: 90%;
    `};
`;

export default Button;
