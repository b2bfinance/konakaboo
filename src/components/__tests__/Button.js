import React from 'react';
import { render } from 'enzyme';
import Button from '../Button';

describe('<Button />', () => {
  test('renders correctly', () => {
    expect(
      render(<Button theme={stubData.theme}>button</Button>)
    ).toMatchSnapshot();
  });
});
