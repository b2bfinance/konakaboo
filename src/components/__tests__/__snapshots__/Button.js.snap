// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Primary button renders correctly 1`] = `
.c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-radius: 0.15rem;
  box-sizing: border-box;
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  font-weight: 400;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 0.75rem 1.5rem;
  text-decoration: none;
  width: 100%;
  white-space: nowrap;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  background-color: #4dc16f;
  color: #fff;
}

.c0:hover {
  text-decoration: none;
}

.c0:hover {
  color: #fff;
}

<a
  class="c0"
>
  
</a>
`;

exports[`Secondary button renders correctly 1`] = `
.c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-radius: 0.15rem;
  box-sizing: border-box;
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  font-weight: 400;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 0.75rem 1.5rem;
  text-decoration: none;
  width: 100%;
  white-space: nowrap;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  background-color: #f5f6f7;
  color: #37474f;
}

.c0:hover {
  text-decoration: none;
}

.c0:hover {
  color: #37474f;
}

<a
  class="c0"
>
  
</a>
`;

exports[`Slim button renders correctly 1`] = `
.c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-radius: 0.15rem;
  box-sizing: border-box;
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  font-weight: 400;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 0.75rem 1.5rem;
  text-decoration: none;
  width: 100%;
  white-space: nowrap;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  padding: 0.25rem 0.5rem;
  font-size: 90%;
}

.c0:hover {
  text-decoration: none;
}

<a
  class="c0"
>
  
</a>
`;
